### PR TITLE
Update key vault SDK  from t1 to t2

### PR DESCRIPTION
**Pull request by LuHang2021**
1. Update key vault SDK  from t1 to t2 
2. Changed files
    1)  azurekeyVaultPollSample/pom.xml
    2)  azurekeyVaultPollSample/src/main/java/com/microsoft/JavaKeyVaultAuthenticator.java
    3)  azurekeyVaultPollSample/src/main/java/com/microsoft/SignVerifySamplesKeyVault.java
    4)  azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java
    5)  azurekeyVaultPollSample/src/main/resources/azure.properties

@jongio for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio Who should we ping to review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @joshfree, @alexGhiondea, @backwind1233  Could you please review this PR?

**Comment by joshfree** [comment]
@chenrujun and @vcolin7 should review this as they own the KV spring boot starter and KV SDK for Java, respectively.

File: azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java [hunk 1]
```diff
@@ -60,104 +54,106 @@ public static void main( String[] args )
...
+        secretAsyncClient.setSecret("secretNameInVault", "secretValue").subscribe(keyVaultSecret -> {
+            System.out.println("The secret value is: " + keyVaultSecret.getValue());
+        });
+
+        Thread.sleep(10000);
+
+        System.out.println("Now listing keys and secrets in Azure Key Vault.");
+
+        keyClient.listPropertiesOfKeys().stream().forEach(keyProperty -> {
+            System.out.printf("key attributes : enabled is %s, notBefore is %s, expires is %s, created is %s, updated is %s \n",
```
Comment (joshfree) [inline azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java:98]: nit: should the printf more closely match the property names?  specificially expires -> ExpiresOn, created -> CreatedOn, and updated -> UpdatedOn

**Review by joshfree — APPROVED** [review]

File: azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java (hunk 1 above)
Comment (hongli750210) [reply azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java]: According to your comment, it has been fixed in the new version .

**Review by vcolin7 — CHANGES_REQUESTED** [review]
Took a good looks and added some questions, comments and suggestions.

File: azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java [hunk 2]
```diff
@@ -1,44 +1,38 @@
...
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 
+
 /**
  * Azure key Vault example,
  * load authentication at runtime,
  * create a key vault and keys and secrets in vault.
  * Continuously poll key vault for keys.
```
[the remark this answers is not in the document]
Comment (hongli750210) [reply azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java]: According to your comment, it has been fixed in the new version .

**Review by vcolin7 — COMMENTED** [review]
Added a few more comments. Thanks for your hard work :)

File: azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java [hunk 3]
```diff
@@ -71,89 +76,92 @@ public static void main( String[] args )
...
+        // Now let's list the existing keys in the vault and use them for signing data and verifying the signature.
+        keyClient.listPropertiesOfKeys().stream().forEach(keyProperties -> {
+            System.out.printf("Key attributes: EnabledOn:%s, NotBeforeOn:%s, ExpiresOn:%s, CreatedOn:%s, UpdatedOn:%s%n",
+                    keyProperties.isEnabled(), keyProperties.getNotBefore(), keyProperties.getExpiresOn(), keyProperties.getCreatedOn(), keyProperties.getUpdatedOn());
+            System.out.println("Key tags are: " + keyProperties.getTags());
+            System.out.println("Key ID: " + keyProperties.getId());
+
+            KeyVaultKey key = keyClient.getKey(keyProperties.getName());
+            String keyName = keyProperties.getName();
+            // Let's sign data using the
```
[the remark this answers is not in the document]
Comment (hongli750210) [reply azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java:117]: According to your comment, it has been fixed in the new version .

**Comment by hongli750210** [comment]
@vcolin7  The comments you added has been fixed in the new version. Thank you for pointing out the deficiencies :)

**Comment by chenrujun** [comment]
> @chenrujun and @vcolin7 should review this as they own the KV spring boot starter and KV SDK for Java, respectively.

Hi, @joshfree , seems this PR does not have dependency of `azure-spring-boot-starter-keyvault-secrets`. So I'll not review this.

**Review by vcolin7 — APPROVED** [review]

**Comment by v-xuto** [comment]
@vcolin7 , @joshfree Could you please help us merge this PR?

**Comment by joshfree** [comment]
> @vcolin7 , @joshfree Could you please help us merge this PR?

I can't, I don't have permission.  Please work with @jongio

**Comment by jongio** [comment]
@v-xuto - Please ensure all conversations are resolved.  If so, click the "Resolve Conversation" button.

**Comment by v-xuto** [comment]
@jongio - We have resolved all the comments, but we do not have permission to click on "Resolve conservation".

**Pull request merged**